src/main/java/Tree/CartesianTree.java

```diff
@@ -0,0 +1,79 @@
+package Tree;
+
+import Node.BinaryNode;
+
+
+public class CartesianTree<T> extends BinaryTree<T> {
+    int[] sequence;
+    int length;
+
+    public CartesianTree() { current = null; }
+
+    public CartesianTree(int[] sequence, int length) {
```

Comment:
I think you should just take in `Boolean maxHeap` here and get rid of the heapTree function